src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -143,28 +144,32 @@ void appendIndentedLines(String s) {
         sb.appendIndentedLines(s);
     }
 
+    void appendIndentedLines(String s, String... args) {
+        sb.appendIndentedLines(String.format(s, (Object []) args));
+    }
+
     final void emitDefaultConstructor() {
-        appendIndentedLines(STR."""
+        appendIndentedLines("""
 
-            \{className}() {
+            %1$s() {
                 // Should not be called directly
             }
-            """);
+            """, className);
     }
 
     final void emitDocComment(Declaration decl) {
         emitDocComment(decl, "");
     }
 
     final void emitDocComment(Declaration decl, String header) {
-        appendLines(STR."""
+        appendLines("""
             /**
-            \{!header.isEmpty() ? STR." * \{header}\n" : ""}\
+            %1$s
              * {@snippet lang=c :
-            \{declarationComment(decl)}
+            %2$s
              * }
              */
-            """);
+            """, !header.isEmpty() ? String.format(" * %1$s\n", header) : "", declarationComment(decl));
```

Comment:
optional comment: we could add a `format` method (which just calls String::format) and then implement `appendIndentedLines` on top. I guess what I wonder is if, at some point, it might make sense to create a more specialized format syntax for jextract use to make `%1$s` a bit more readable.